app/views/spree/components/buttons/_button_primary.html.erb

```diff
@@ -11,11 +11,12 @@
   type = local_assigns.fetch(:type, 'button')
 
   # Classes
-  class_names = [base_class]
-  class_names.push("#{base_class}--bordered") if bordered
-  class_names.push("#{base_class}--full-width") if full_width
-  class_names.push("#{base_class}--small") if small
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
```

Comment:
I think that we could move this line above in the _Optional props_ list to preserve the logic (the same comment is valid for the whole PR).